samsa2-master.nf

```diff
@@ -0,0 +1,258 @@
+
+    output:
+    tuple val(sample_id), path("${sample_id}.merged.Subsys_annotated"), path("${ribodepleted_fastq}.Subsys.daa")
+
+    script:
+    """
+    ${params.programs}/diamond blastx --db $params.subsys_database -q $ribodepleted_fastq -a ${ribodepleted_fastq}.Subsys -t /vast/scratch/users/\$USER/tmp -k 1 -p $task.cpus -b 12 -c 1
+    ${params.programs}/diamond view --daa ${ribodepleted_fastq}.Subsys.daa -o ${sample_id}.merged.Subsys_annotated -f tab -p $task.cpus
+    """
+
+
+}
+
+process REFSEQ_ANALYSISCOUNTER_FUNC {
+
+    cpus 46
+    memory '64 GB'
+    publishDir "${step_5_output_dir}/RefSeq_results/func_results", mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(refseq_annotated)
+
+    output:
+    tuple val(sample_id), path("${sample_id}.merged.*.tsv")
+
+    script:
+    """
+    python ${params.python_scripts}/DIAMOND_analysis_counter_mp.py -I ${refseq_annotated} -D $refseq_db -F -t $task.cpus
+    """
+}
+
+process REFSEQ_ANALYSISCOUNTER_ORG {
+
+    cpus 46
+    memory '64 GB'
+    publishDir "${step_5_output_dir}/RefSeq_results/org_results", mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(refseq_annotated)
+
+    output:
+    tuple val(sample_id), path("${sample_id}.merged.*.tsv")
+
+    script:
+    """
+    python ${params.python_scripts}/DIAMOND_analysis_counter_mp.py -I ${refseq_annotated} -D $refseq_db -O -t $task.cpus
+    """
+}
+
+process SUBSYS_ANALYSIS_COUNTER {
+
+    cpus 2
+    memory '64 GB'
+    publishDir "${step_5_output_dir}/Subsystems_results/receipts", pattern: '*.receipt', mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(subsys_annotated)
+
+    output:
+    tuple val(sample_id), path('*.hierarchy'), path('*.receipt')
+
+    script:
+    """
+    python ${params.python_scripts}/DIAMOND_subsystems_analysis_counter.py \
+        -I $subsys_annotated \
+        -D ${params.subsys_database}.fa \
+        -O ${subsys_annotated}.hierarchy \
+        -P ${subsys_annotated}.receipt
+    """
+
+}
+
+process SUBSYS_REDUCER {
+
+    cpus 2
+    memory '64 GB'
+    publishDir "${step_5_output_dir}/Subsystems_results", mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(subsys_annotated_hierarchy)
+
+    output:
+    tuple val(sample_id), path('*.reduced')
+
+    script:
+    """
+    python ${params.python_scripts}/subsys_reducer.py -I $subsys_annotated_hierarchy
+    """
+}
+
+// process R {
+
+// }
+
+workflow {
+
+
+    infiles_ch = Channel.fromPath(input_files)
+
+    Channel
+        .fromFilePairs(input_files, checkIfExists: true)
+        .set { read_pairs_ch }
+
+    trim_ch = TRIMMOMATIC(read_pairs_ch)
+
+    pear_ch = PEAR(trim_ch)
+
+    forward_reads_ch = trim_ch.map{it[1]}.collect(flat: false)
+    raw_counts_ch = RAWREADCOUNT(forward_reads_ch)
+
+    sortmerna_input_ch = pear_ch.map{[it[0],it[1]]}
+    sortmerna_ch = SORTMERNA(sortmerna_input_ch)
+
+    // REFSEQ
+    diamond_refseq_ch = DIAMOND_REFSEQ(sortmerna_ch)
```

Comment:
You can use the [pipe](https://www.nextflow.io/docs/latest/dsl2.html#the-pipe-operator) to avoid needing to create intermediate variables. But this is also just style/preference.

Reply:
opened as issue https://github.com/WEHI-ResearchComputing/samsa2-nf/issues/2